tests/unit/test_charm_2.py

```diff
@@ -1,58 +1,166 @@
+import re
+import json
 import unittest
+import collections
 import unittest.mock
+
 import ops.pebble
 import ops.testing
+import mysql.connector
 
 from charm import WordpressCharm
 from exceptions import *
 
 
-class TestWordpressK8s(unittest.TestCase):
-    def setUp(self):
-        container_fs = {}
-        self.container_fs = container_fs
-        installed_database_hosts = set()
-        self.installed_database_hosts = installed_database_hosts
+class WordpressMock:
+    """WordPress wp-cli command run and database simulation system for unit tests"""
+    def __init__(self):
+        self._database = {}
+        self._container_fs = {}
+        self._database_credentials = {}
+        self._themes = set(WordpressCharm._WORDPRESS_DEFAULT_THEMES)
+        self._plugins = set(WordpressCharm._WORDPRESS_DEFAULT_PLUGINS)
+        self._patch = None
+
+    def _get_current_database_config(self):
+        wp_config = self._container_fs.get(WordpressCharm._WP_CONFIG_PATH)
+        if wp_config is None:
+            return None
+        db_info = {}
+        for db_key in ['db_host', 'db_name', 'db_user', 'db_password']:
```

Comment:
Picky comment, do you think a tuple is more appropriate here since things like being able to resize the list are not needed here?

Reply:
Fixed, thanks!